ssl/record/rec_layer_s3.c

```diff
@@ -1279,6 +1279,22 @@ int ssl3_write_pending(SSL_CONNECTION *s, int type, const unsigned char *buf,
     }
 }
 
+int ssl_auto_retry(SSL *s)
```

Comment:
As this is internal function it would be better to use `SSL_CONNECTION *` as parameter here - all the callers seem to already have it. You can get the `SSL *` then by `SSL_CONNECTION_GET_SSL()`. Result of `SSL_CONNECTION_FROM_SSL()` should be checked for NULL where the `SSL_CONNECTION_GET_SSL()` can never fail on proper SSL_CONNECTION parameter.